orchestrator/orchestrator/src/ethereum_event_watcher.rs

```diff
@@ -223,20 +223,26 @@ pub async fn check_for_events<S: Signer + 'static, CS: CosmosSigner>(
             .await
             .expect("Could not send messages");
 
+
+        let mut error_count: u32 = 0;
         let timeout = time::Duration::from_secs(30);
         contact.wait_for_next_block(timeout).await?;
-
-        // TODO(bolten): we are only waiting one block, is it possible if we are sending multiple
-        // events via the sender, they could be received over the block boundary and thus our new
-        // event nonce does not reflect full processing of the above events?
-        let new_event_nonce = get_last_event_nonce(grpc_client, our_cosmos_address).await?;
-        if new_event_nonce == last_event_nonce {
-            return Err(GravityError::InvalidBridgeStateError(
-                format!("Claims did not process, trying to update but still on {}, trying again in a moment", last_event_nonce),
-            ));
-        } else {
-            info!("Claims processed, new nonce: {}", new_event_nonce);
+        let mut new_event_nonce = get_last_event_nonce(grpc_client, our_cosmos_address).await?;
```

Comment:
Modified this part to have a stronger check on the nonce.

instead of just checking that the nonce is different from previous, we are strictly checking equality with the last message nonce otherwise we don't let the orchestrator continue.

If after 10 blocks, it is still failing then we replay this block section